src/Apps/MyCollection/Routes/EditArtwork/MyCollectionEditArtwork.tsx

```diff
@@ -0,0 +1,181 @@
+
+      await Promise.all(
+        removedPhotos.map(async photo => {
+          try {
+            await deleteArtworkImage({
+              variables: {
+                input: {
+                  artworkID: artwork?.internalID!,
+                  imageID: photo.internalID!,
+                },
+              },
+            })
+          } catch (error) {
+            logger.error("An error occured while removing images.", error)
+
+            sendToast({
+              variant: "error",
+              message: "An error occurred",
+              description: "Couldn't remove all images.",
+            })
+          }
+        })
+      )
+
+      router.replace({ pathname: `/settings/my-collection` })
+      router.push({ pathname: `/my-collection/artwork/${artworkId}` })
+    } catch (error) {
+      logger.error(`Artwork not updated`, error)
+
+      sendToast({
+        variant: "error",
+        message: "An error occurred",
+        description: "Please contact support@artsymail.com",
+      })
+    }
+  }
+
+  const handleBack = () => {
+    router.push({
+      pathname: `/my-collection/artwork/${artwork?.internalID}`,
+    })
+  }
+
+  return (
+    <>
+      <MetaTags
+        title={`Edit ${artwork?.title} - ${artwork?.artistNames} | Artsy`}
+      />
+
+      <MyCollectionArtworkFormContextProvider
+        artworkFormImagesRef={artworkFormImagesRef}
+        onBack={handleBack}
+      >
+        <Formik<ArtworkModel>
+          validateOnMount
+          onSubmit={handleSubmit}
+          initialValues={getMyCollectionArtworkFormInitialValues(artwork)}
+          validationSchema={MyCollectionArtworkDetailsValidationSchema}
+        >
+          <MyCollectionArtworkFormMainFragmentContainer artwork={artwork} />
+        </Formik>
+      </MyCollectionArtworkFormContextProvider>
+    </>
+  )
+}
+
+export const MyCollectionEditArtworkFragmentContainer = createFragmentContainer(
+  MyCollectionEditArtwork,
+  {
+    artwork: graphql`
+      fragment MyCollectionEditArtwork_artwork on Artwork {
+        artist {
+          internalID
+          name
+          formattedNationalityAndBirthday
+          targetSupply {
+            isP1
+          }
+        }
+        consignmentSubmission {
+          inProgress
+        }
+        artistNames
+        category
+        pricePaid {
+          display
+          minor
+          currencyCode
+        }
+        date
+        depth
+        dimensions {
+          in
+          cm
+        }
+        editionSize
+        editionNumber
+        height
+        attributionClass {
+          name
+        }
+        id
+        images {
+          internalID
+          isDefault
+          imageURL
+          width
+          height
+          internalID
+        }
+        internalID
+        isEdition
+        medium
+        metric
+        artworkLocation
+        provenance
+        slug
+        title
+        width
```

Comment:
**thought:** I am wondering if there is a way we can avoid this. Have you considered using `@relay(mask: false)` or actually just renaming `artwork` to something else and injecting `MyCollectionArtworkFormMain_artwork` there. My goal here is to avoid forgetting adding params here if we add them to the form.

_Probably this is not needed here since this is existing code, or you might be already doing some dynamic checking in the mutation_

Reply:
The fields are specified here because this is the place where the form gets initiated. `MyCollectionArtworkFormMain` just needs `internalId` and `slug`. Not sure if we could or should move the fields to another place.

Reply:
aha I get you. It's probably not possible 😄. Thanks for explaining